tests/vmi_hook_sidecar_test.go

```diff
@@ -165,6 +171,127 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
+		}
+
+		Context("with QEMU Args hook sidecar", func() {
+			It("should successfully start with hook sidecar annotation", func() {
+				By("Starting a VMI")
+				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				Expect(err).ToNot(HaveOccurred())
+				libwait.WaitForSuccessfulVMIStart(vmi)
+			})
+
+			It("should successfully start with hook sidecar annotation for v1alpha2", func() {
+				By("Starting a VMI")
+				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				vmi.ObjectMeta.Annotations = RenderValidQEMUArgsSidecar(hooksv1alpha2.Version)
+				Expect(err).ToNot(HaveOccurred())
+				libwait.WaitForSuccessfulVMIStart(vmi)
+			})
+
+			It("should call Collect and OnDefineDomain on the hook sidecar", func() {
+				By("Getting hook-sidecar logs")
+				vmi, err = virtClient.
+					VirtualMachineInstance(testsuite.GetTestNamespace(nil)).
+					Create(context.Background(), vmi)
+				Expect(err).ToNot(HaveOccurred())
+				logs := func() string { return getHookSidecarLogs(virtClient, vmi) }
+				libwait.WaitForSuccessfulVMIStart(vmi)
+				Eventually(logs,
+					11*time.Second,
+					500*time.Millisecond).
+					Should(ContainSubstring("Hook's Info callback method has been called"))
+				Eventually(logs,
+					11*time.Second,
+					500*time.Millisecond).
+					Should(ContainSubstring("Hook's OnDefineDomain callback method has been called"))
+			})
+
+			It("should update domain XML with QEMU command line options and environment variables", func() {
+				By("Reading domain XML using virsh")
+				clientcmd.SkipIfNoCmd("kubectl")
+
+				strategy := v1.StartStrategyPaused
+				vmi.Spec.StartStrategy = &strategy
+				vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
+				Expect(err).ToNot(HaveOccurred(), "Create VMI successfully")
+				libwait.WaitForSuccessfulVMIStart(vmi)
+				Eventually(matcher.ThisVMI(vmi), 30*time.Second, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
+
+				domainXml, _ := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+				fmt.Println(domainXml)
+				Expect(domainXml).Should(ContainSubstring("<qemu:commandline>"))
+				Expect(domainXml).Should(ContainSubstring("<qemu:env name='G_MESSAGES_DEBUG' value='all'/>"))
+				Expect(domainXml).Should(ContainSubstring("</qemu:commandline>"))
+			})
+
+			It("should not start with hook sidecar annotation when the version is not provided", func() {
+				By("Starting a VMI")
+				vmi.ObjectMeta.Annotations = RenderInvalidQEMUArgsSidecar("lacks version")
+				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				Expect(err).NotTo(HaveOccurred(), "the request to create the VMI should be accepted")
+
+				Eventually(func() bool {
+					vmiPod, exists, err := getVMIPod(vmi)
+					if err != nil {
```

Comment:
You don't really need to check the err in the if statement if it is followed by `Expect(err).ToNot(HaveOccurred())`

Reply:
Fixed, thanks!